R/eval.R

```diff
@@ -96,26 +96,22 @@ evaluate <- function(input,
     )
     watcher$check_devices()
 
-    if (stop_on_error > 0L) {
-      errs <- vapply(out[[i]], is.error, logical(1))
-
-      if (!any(errs)) next
-      if (stop_on_error == 1L) break
+    if (stop_on_error == 1L && output$has_errored()) {
```

Comment:
The previous logic was rather hard to parse, but I think this is what it's getting at.